kmir/k-src/mir-rvalue.md

```diff
@@ -364,6 +369,23 @@ Locals only makes sense within a function-like, hence we evaluate them as a cont
     requires PLACE_INDEX ==Int Local2Int(PLACE)
 ```
 
+### `Enum` evaluation
+```k
+  syntax MIRValue ::= evalEnumConstructor(FunctionLikeKey, EnumConstructor) [function]
+  //--------------------------------------------------------------
+  rule evalEnumConstructor(FN_KEY, Option :: < _TYPES > :: .ExpressionPathList :: Some ( OP , .OperandList ) ) => OptSome(evalOperand(FN_KEY, OP:Operand))
+    requires IdentifierToken2String(Option) ==String "Option" andBool IdentifierToken2String(Some) ==String "Some"
```

Comment:
@Baltoli I couldn't find it in the documentation for K, but is there a way to be have uppercase sequences of characters not match as a variable. To be explicit `Option` above is a variable, but this rule is expecting that variable is actually only `Option` verbatim, which is added as a side condition. Wrapping in `""` creates a `String`. Can I signal that I want the upper case to be treated verbatim?

Reply:
You want to declare `Option` as a token production of sort `IdentifierToken`; it will then parse as a token rather than as a variable. This is alluded to in Lesson 16 of the K tutorial, but not very clearly at all so I think we could probably improve the documentation there.

The issue is that the **rule parser** behaves differently to the program parser in a few ways to make it easier to write common-case rewrite rules. One of these is in the overlap between scanning sequences of characters; anything that can be scanned as a variable will be, under the assumption that users are more likely to write a rule mentioning variables than they are to mention an upper-cased literal token in some sort.

Here's a K definition demonstrating this difference:
```k
module UPPER-SYNTAX
  imports ID-SYNTAX

  syntax Foo ::= "foo" Id
               | "bar" Id
endmodule

module UPPER
  imports UPPER-SYNTAX

  syntax Id ::= "Option"  [token]
              | "Some"    [token]

  rule foo Option => bar Some [label(token)]
  rule foo A => bar A         [label(variable), owise]
endmodule
```